File: applications/heatpump-web/src/types/heatpump.ts
export interface HeatpumpStatus {
  ts: string;
  device_id: string | null;
  compressor_on: boolean | null;
  hotwater_production: boolean | null;
  flowlinepump_on: boolean | null;
  brinepump_on: boolean | null;
  aux_heater_3kw_on: boolean | null;
  aux_heater_6kw_on: boolean | null;
  outdoor_temp: number | null;
  supplyline_temp: number | null;
  returnline_temp: number | null;
  hotwater_temp: number | null;
  brine_out_temp: number | null;
  brine_in_temp: number | null;
}

File: applications/heatpump-web-leptos/src/models/heatpump.rs
use serde::{Deserialize, Serialize};

/// Heatpump status from the monitoring system
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct HeatpumpStatus {
    pub ts: String,
    pub device_id: Option<String>,
    pub compressor_on: Option<bool>,
    pub hotwater_production: Option<bool>,
    pub flowlinepump_on: Option<bool>,
    pub brinepump_on: Option<bool>,
    pub aux_heater_3kw_on: Option<bool>,
    pub aux_heater_6kw_on: Option<bool>,
    pub outdoor_temp: Option<f64>,
    pub supplyline_temp: Option<f64>,
    pub returnline_temp: Option<f64>,
    pub hotwater_temp: Option<f64>,
    pub brine_out_temp: Option<f64>,
    pub brine_in_temp: Option<f64>,
    pub integral: Option<f64>,
}
